code/game/objects/items/devices/scanners.dm

```diff
@@ -386,67 +384,6 @@ MASS SPECTROMETER
 		to_chat(user, "<span class='info'>Temperature: [round(environment.temperature-T0C)] &deg;C</span>")
 
 
-/obj/item/device/mass_spectrometer
-	desc = "A hand-held mass spectrometer which identifies trace chemicals in a blood sample."
-	name = "mass-spectrometer"
-	icon_state = "spectrometer"
-	item_state = "analyzer"
-	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
-	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
-	w_class = WEIGHT_CLASS_SMALL
-	flags_1 = CONDUCT_1
-	slot_flags = SLOT_BELT
-	container_type = OPENCONTAINER_1
-	throwforce = 0
-	throw_speed = 3
-	throw_range = 7
-	materials = list(MAT_METAL=150, MAT_GLASS=100)
-	var/details = 0
-
-/obj/item/device/mass_spectrometer/New()
-	..()
-	create_reagents(5)
-
-/obj/item/device/mass_spectrometer/on_reagent_change()
-	if(reagents.total_volume)
-		icon_state = initial(icon_state) + "_s"
-	else
-		icon_state = initial(icon_state)
-
-/obj/item/device/mass_spectrometer/attack_self(mob/user)
-	if (user.stat || user.eye_blind)
-		return
-	if (!user.IsAdvancedToolUser())
-		to_chat(user, "<span class='warning'>You don't have the dexterity to do this!</span>")
-		return
-	if(reagents.total_volume)
-		var/list/blood_traces = list()
-		for(var/datum/reagent/R in reagents.reagent_list)
-			if(R.id != "blood")
-				reagents.clear_reagents()
-				to_chat(user, "<span class='warning'>The sample was contaminated! Please insert another sample.</span>")
-				return
-			else
-				blood_traces = params2list(R.data["trace_chem"])
-				break
-		var/dat = "<i><b>Trace Chemicals Found:</b>"

```

Comment:
AFAIK the health sensor DOES NOT recreate this functionality(scanning a blood sample for what chems **used** to be in it), maybe just add this behavior to the detective scanner? It's still useful for a detective to try and determine if someone was poisoned, even after all that poison might be out of the body.

Reply:
Trace chems store the set of chems that were in the body when the blood sample was taken. If the toxin is out of the body already, it wouldn't help.